### PR TITLE
Fix docs and code to match in haystack tracing.

### PR DESCRIPTION
**Pull request by evanlurvey**
### What does this PR do?

This PR addresses documentation errors and a default value change. The docs stated that defaults were one way but the code was defined otherwise. Sensible defaults were also chosen in the docs over "sample".

### Motivation

This PR was motivated by consuming the docs when integrating with haystack and getting an unexpected outcome.

### More

- [ ] Added/updated tests
- [x] Added/updated documentation

### Additional Notes

N/A

### CODE REVIEW

File: pkg/tracing/haystack/haystack.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ type Config struct {
 
 // SetDefaults sets the default values.
 func (c *Config) SetDefaults() {
-	c.LocalAgentHost = "LocalAgentHost"
+	c.LocalAgentHost = "127.0.0.1"
```
Comment (mpl) [inline pkg/tracing/haystack/haystack.go]: Can you please explain as if we did not know anything about haystack (which is the case for most of us), what is the rationale behind that change?

Comment (evanlurvey) [reply pkg/tracing/haystack/haystack.go]: Sure, Haystack is a tracing tool designed by Expedia Group. To forward metrics to haystack you need a [haystack-agent](https://github.com/ExpediaDotCom/haystack-agent). The most common way to run this agent is as a docker side car to your application. When configured this way the agent will be listing on port 35000 by default and be associated with the same loopback address as your main service (traefik in this case). Also in the existing documentation for traefik it states that the default is 127.0.0.1 however I think it was accidentally set to LocalAgentHost.

Comment (ldez) [reply pkg/tracing/haystack/haystack.go]: ping @aantono

**Comment by ldez** [comment]
@evanlurvey  thanks for your contribution.

You don't need to "merge v2.1", we have a bot for that :wink:

**Pull request merged**